tensorflow/contrib/tensorrt/python/nvidia_scope.py

```diff
@@ -0,0 +1,200 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+"""Exposes the scope for NVIDIA optimizers."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import contextlib
+
+from tensorflow.core.framework import attr_value_pb2
+from tensorflow.python.framework import ops
+
+# follows xla_scope as example
+
+_NV_SCOPE_KEY = ("__nv_scope",)
+
+
+class NvidiaScope(object):
+  """Keeps track of scopes"""
+
+  def __init__(self, depth):
+    self.depth = depth
+
+
+def ConditionalWrapper(selector, value):
+  """ Simple wrapper that returns a callable to disable attributes for the excluded nodes."""
+
+  def ConditionalSetter(node_def):
+    if callable(selector):
+      accept = selector(node_def).b
+    else:
+      accept = selector.b
+    if accept:
+      return value
+    return None
+
+  return ConditionalSetter
+
+
+@contextlib.contextmanager
+def nvidia_scope(
+    select=True,
+    scope_name=None,
+    optimizer=None,
+    precision=None,
+    engine_mode=None,
+    maximum_cached_engines=None,
+    maximum_workspace_size=None,
+):
+  """Include or exclude nodes in optimized segment with given parameters.
+
+  NOTE: this is a hint and will be supported on a best-effort basis.
+  Args:
+    include      : Whether to include the node or not.
```

Comment:
Looks like you have renamed `include` to `select` in the code but forgot to rename it in the comments?